src/main/java/com/google/sps/data/Journey.java

```diff
@@ -79,45 +87,181 @@ public int getNumberOfWaypoints() {
   /**
    * Returns true if the minimum time journey fits in the timeslot and false otherwise.
    */
-  public boolean findJourneyForTimeslot(DeliverySlot deliverySlot) throws ApiException, IOException, InterruptedException {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public boolean findJourneyForTimeslot(DeliverySlot deliverySlot) throws ApiException, DataNotFoundException, IOException, InterruptedException {
+    ArrayList<ArrayList<Pair>> bestTime = findOptimalTimePaths();
+    int numberOfWaypoints = getNumberOfWaypoints();
+    int allWaypointsConfig = (1 << numberOfWaypoints) - 1;
```

Comment:
I'll have a look on that, thanks! 
The reason why the limit is 25 is because the time complexity for optimizing one journey is quite bad : O((2^(numberOfWaypoints)) * numberOfWaypoints)